src/crypto/tls/delegated_credentials.go

```diff
@@ -296,20 +303,22 @@ func getSignatureAlgorithm(cert *Certificate) (SignatureScheme, error) {
 	case *ecdsa.PrivateKey:
 		pk := sk.Public().(*ecdsa.PublicKey)
 		curveName := pk.Curve.Params().Name
-		certAlg := cert.Leaf.SignatureAlgorithm
-		if certAlg == x509.ECDSAWithSHA256 && curveName == "P-256" {
+		certAlg := cert.Leaf.PublicKeyAlgorithm
+		if certAlg == x509.ECDSA && curveName == "P-256" {
 			sigAlgo = ECDSAWithP256AndSHA256
-		} else if certAlg == x509.ECDSAWithSHA384 && curveName == "P-384" {
+		} else if certAlg == x509.ECDSA && curveName == "P-384" {
 			sigAlgo = ECDSAWithP384AndSHA384
-		} else if certAlg == x509.ECDSAWithSHA512 && curveName == "P-521" {
+		} else if certAlg == x509.ECDSA && curveName == "P-521" {
 			sigAlgo = ECDSAWithP521AndSHA512
 		} else {
 			return SignatureScheme(0x00), fmt.Errorf("using curve %s for %s is not supported", curveName, cert.Leaf.SignatureAlgorithm)
 		}
 	case ed25519.PrivateKey:
 		sigAlgo = Ed25519
+	case *rsa.PrivateKey:
+		return PSSWithSHA256, nil
 	default:
-		return SignatureScheme(0x00), fmt.Errorf("tls: unsupported algorithm for Delegated Credential")
+		return SignatureScheme(0x00), fmt.Errorf("tls: unsupported algorithm for signing Delegated Credential")
```

Comment:
`undefinedSignatureScheme` would also be useful here. Alternatively, you could just return `sigAlgo`, which defaults to `0x0000`.